src/tests/unit/platform/test_policy_ruleconfigs.py

```diff
@@ -0,0 +1,296 @@
+                             CORE_PREVENTION_RETURNS)
+    api = cbcsdk_mock.api
+    policy = Policy(api, 65536, copy.deepcopy(FULL_POLICY_1), False, True)
+    rule_config = random.choice([cfg for cfg in policy.object_rule_configs.values()
+                                if isinstance(cfg, CorePreventionRuleConfig)])
+    rule_config.refresh()
+
+
+def test_core_prevention_set_assignment_mode(cb):
+    """Tests the assignment mode setting, which uses the underlying parameter setting."""
+    policy = Policy(cb, 65536, copy.deepcopy(FULL_POLICY_1), False, True)
+    rule_config = random.choice([cfg for cfg in policy.object_rule_configs.values()
+                                 if isinstance(cfg, CorePreventionRuleConfig)])
+    old_mode = rule_config.get_assignment_mode()
+    assert not rule_config.is_dirty()
+    rule_config.set_assignment_mode(old_mode)
+    assert not rule_config.is_dirty()
+    rule_config.set_assignment_mode('BLOCK' if old_mode == 'REPORT' else 'REPORT')
+    assert rule_config.is_dirty()
+    with pytest.raises(ApiError):
+        rule_config.set_assignment_mode('BOGUSVALUE')
+
+
+def test_core_prevention_update_and_save(cbcsdk_mock):
+    """Tests updating the core prevention data and saving it."""
+    put_called = False
+
+    def on_put(url, body, **kwargs):
+        nonlocal put_called
+        assert body == CORE_PREVENTION_UPDATE_1
+        put_called = True
+        return copy.deepcopy(body)
+
+    cbcsdk_mock.mock_request('GET', '/policyservice/v1/orgs/test/policies/65536/configs/presentation',
+                             POLICY_CONFIG_PRESENTATION)
+    cbcsdk_mock.mock_request('PUT', '/policyservice/v1/orgs/test/policies/65536/rule_configs/core_prevention', on_put)
+    api = cbcsdk_mock.api
+    policy = Policy(api, 65536, copy.deepcopy(FULL_POLICY_1), False, True)
+    rule_config = policy.object_rule_configs['c4ed61b3-d5aa-41a9-814f-0f277451532b']
+    assert rule_config.name == 'Carbon Black Threat Intel'
+    assert rule_config.get_assignment_mode() == 'REPORT'
+    rule_config.set_assignment_mode('BLOCK')
+    rule_config.save()
+    assert put_called
+
+
+def test_core_prevention_update_via_replace(cbcsdk_mock):
+    """Tests updating the core prevention data and saving it via replace_rule_config."""
+    put_called = False
+
+    def on_put(url, body, **kwargs):
+        nonlocal put_called
+        assert body == CORE_PREVENTION_UPDATE_1
+        put_called = True
+        return copy.deepcopy(body)
+
+    cbcsdk_mock.mock_request('GET', '/policyservice/v1/orgs/test/policies/65536/configs/presentation',
+                             POLICY_CONFIG_PRESENTATION)
+    cbcsdk_mock.mock_request('PUT', '/policyservice/v1/orgs/test/policies/65536/rule_configs/core_prevention', on_put)
+    api = cbcsdk_mock.api
+    policy = Policy(api, 65536, copy.deepcopy(FULL_POLICY_1), False, True)
+    rule_config = policy.object_rule_configs['c4ed61b3-d5aa-41a9-814f-0f277451532b']
```

Comment:
This pattern for fetching a specific rule config makes a challenging experience if the user doesn't have the ids of the specific rule configs for that category. Could we make a core_prevention helper function which fetches all the core_prevention configs at the policy level?

Reply:
I think it would be worth considering what the normal usage is going to be for the CBC SDK and the rule configs. How are users going to lookup rule configs? What are they going to do with it once they have the rule config? Let's discuss this in a meeting next week to identify some usage patterns

Reply:
I'm certainly willing to add the helper function at the policy level; it's pretty much a one-liner, as far as I can tell.

Reply:
Please consider the usability of the added core prevention functionaity and how it will be consumed by a developer

Reply:
You're probably right that we need to just sit down and talk about this part, because I can't read your mind and see what _you_ mean by "usability...and how it will be consumed by a developer."